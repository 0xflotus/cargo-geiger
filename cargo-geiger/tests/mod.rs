use assert_cmd::prelude::*;
use insta::assert_snapshot_matches;
use rstest::rstest_parametrize;

use std::env;
use std::path::Path;
use std::process::Command;

#[rstest_parametrize(name,
    case("test1_package_with_no_deps"),
    case("test2_package_with_shallow_deps"),
    case("test3_package_with_nested_deps"),
    case("test4_workspace_with_top_level_package"),
    case("test5_workspace_with_virtual_manifest"),
    case("test6_cargo_lock_out_of_date")
)]
fn test_package(name: &str) {
    better_panic::install();

    let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
    let test_case_root_dir = Path::new(&manifest_dir)
        .join("../test_crates")
        .join(name);

    let result = Command::cargo_bin("cargo-geiger")
        .unwrap()
        .arg("geiger")
        .arg("--color=never")
        .arg("--quiet=true")
        .arg("--charset=ascii")
        .arg("--all-targets")
        .arg("--all-features")
        .current_dir(test_case_root_dir)
        .output()
        .expect("failed to run `cargo-geiger`");
   
    let stderr_filename = format!("{}.stderr", name);
    let stderr = String::from_utf8(result.stderr)
        .expect("output should have been valid utf-8");
   
    if !stderr.is_empty() {
        let re = regex::Regex::new(r"`([^`]+).toml`").unwrap();
        let stderr = re.replace(&stderr, "`{MANIFEST_PATH}`");
        assert_snapshot_matches!(stderr_filename, stderr);
    }

    let stdout_filename = format!("{}.stdout", name);
    let stdout = String::from_utf8(result.stdout)
        .expect("output should have been valid utf-8");
    assert_snapshot_matches!(stdout_filename, stdout);

    if stderr.is_empty() {
        assert!(result.status.success(), "`cargo-geiger` failed");
    }
}

