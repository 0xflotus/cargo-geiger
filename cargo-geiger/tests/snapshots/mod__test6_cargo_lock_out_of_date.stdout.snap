---
created: "2019-08-11T20:44:05.368283209Z"
creator: insta@0.10.0
source: cargo-geiger/tests/mod.rs
expression: stdout
---

Metric output format: x/y
    x = unsafe code used by the build
    y = total unsafe code found in the crate

Symbols: 
    :) = No `unsafe` usage found, declares #![forbid(unsafe_code)]
    ?  = No `unsafe` usage found, missing #![forbid(unsafe_code)]
    !  = `unsafe` usage found

Functions  Expressions  Impls  Traits  Methods  Dependency

0/0        0/0          0/0    0/0     0/0      :) geiger-test 0.1.0
0/0        0/0          0/0    0/0     0/0      :) |-- generational-arena 0.2.2
0/0        0/0          0/0    0/0     0/0      ?  |   `-- cfg-if 0.1.9
0/0        1/1          0/0    0/0     0/0      !  `-- idna 0.1.5
0/0        0/0          0/0    0/0     0/0      ?      |-- matches 0.1.8
0/0        0/0          0/0    0/0     0/0      :)     |-- unicode-bidi 0.3.4
0/0        0/0          0/0    0/0     0/0      ?      |   `-- matches 0.1.8
0/0        20/20        0/0    0/0     0/0      !      `-- unicode-normalization 0.1.8
2/2        318/318      4/4    1/1     13/13    !          `-- smallvec 0.6.9

